contracts/governance/GoodDollarStaking.sol

```diff
@@ -244,17 +244,36 @@ contract GoodDollarStaking is
 		return getProductivity(address(this), _user);
 	}
 
-	/// @dev helper function for multibase
-	function getUserPendingReward(address _user) public view returns (uint256) {
-		return getUserPendingReward(address(this), 0, block.number, _user);
-	}
-
 	/// @dev helper function for multibase
 	function goodStakerInfo(address _user) public view returns (UserInfo memory) {
 		return contractToUsers[address(this)][_user];
 	}
 
-	function totalRewardsPerShare() public view returns (uint256) {
-		return totalRewardsPerShare(address(this));
+  /// @dev helper function for multibase and FixedAPYRewards
+	function getUserPendingReward(address _user)
+		public
+		returns (uint256 _goodReward, uint128 _gdReward)
+	{
+		// GOOD rewards
+    _goodReward = getUserPendingReward(
+			address(this),
+			0,
+			block.number,
+			_user
+		);
+    
+    // g$ rewards
+    _updateReward(_user);
+    _gdReward = stakersInfo[_user].reward;
```

Comment:
Could be done just with earned() function, if we revert previous change in FixedAPY

stakersInfo[_account].reward **+**= uint128(earned(_account)); // added +
earned(){ .... calculation... ~~+ stakersInfo[_account].reward~~

but not a big deal

Reply:
you can not simply convert a "view" function into a non-view function
you need to calculate what is the user pending reward but not update it

Reply:
Understood, thx